core/src/main/java/com/styra/opa/wasm/OpaPolicy.java

```diff
@@ -93,6 +93,11 @@ public OpaPolicy data(JsonNode data) {
     }
 
     public OpaPolicy input(String input) {
+        if (this.dataAddr == -1) {
+            // to keep the ordering: data - input - evaluate
+            data("");
```

Comment:
Seems like the ordering of `data - input` is significant, and this makes it more consistent in real world usage.
Happy to hear about a more principled approach :slightly_smiling_face:

Reply:
That's a bit surprising to me, but I know next to nothing about how the Wasm evaluation works. Data should always be an object though, so `{}` would be more appropriate, I think.